src/models/__tests__/profile.test.ts

```diff
@@ -184,147 +187,176 @@ describe("Profile codec", () => {
+      items: {
+        create: jest.fn().mockReturnValue(
+          // this scenario is unlikely to happen because the cosmos SDK should reject the promise
+          // if something went wrong
+          Promise.resolve({
+            resource: undefined
+          })
+        )
+      }
+    } as unknown) as Container;
+
+    const model = new ProfileModel(containerMock);
 
-    const result = await model.create(newProfile, newProfile.fiscalCode);
+    const result = await model.create(newProfile)();
 
-    expect(clientMock.createDocument).toHaveBeenCalledTimes(1);
+    expect(containerMock.items.create).toHaveBeenCalledTimes(1);
 
-    expect(isLeft(result)).toBeTruthy();
-    if (isLeft(result)) {
-      expect(result.value).toEqual("error");
+    expect(E.isLeft(result)).toBeTruthy();
+    if (E.isLeft(result)) {
+      expect(result.left.kind).toEqual("COSMOS_EMPTY_RESPONSE");
     }
   });
 });
 
-describe("update", () => {
+describe("updateProfile", () => {
   it("should update an existing profile", async () => {
-    const clientMock: any = {
-      createDocument: jest.fn((_, newDocument, __, cb) => {
-        cb(undefined, {
-          ...newDocument,
-          _self: "self",
-          _ts: 123
-        });
-      }),
-      readDocument: jest.fn((_, __, cb) => cb(undefined, aRetrievedProfile))
+    const containerMock = ({
+      items: {
+        create: jest.fn().mockReturnValue(
+          Promise.resolve({
+            resource: {
+              ...aRetrievedProfile,
+              id: `${aFiscalCode}-${"0".repeat(15)}1`,
+              version: 1,
+              email: anEmail
+            }
+          })
+        )
+      }
+    } as unknown) as Container;
+
+    const model = new ProfileModel(containerMock);
+
+    const aProfileWithDifferentEmail = {
+      ...aRetrievedProfile,
+      email: anEmail
     };
 
-    const model = new ProfileModel(clientMock, profilesCollectionUrl);
+    const aRetrievedUpdatedProfile = {
+      ...aRetrievedProfile,
+      email: anEmail,
+      id: `${aFiscalCode}-${"0".repeat(15)}1`,
+      version: 1
+    };
 
-    const result = await model.update(
-      aRetrievedProfile.fiscalCode,
-      aRetrievedProfile.fiscalCode,
-      p => {
-        return {
-          ...p,
-          email: "new@example.com" as EmailString
-        };
-      }
-    );
+    const result = await model.update(aProfileWithDifferentEmail)();
 
-    expect(clientMock.createDocument).toHaveBeenCalledTimes(1);
-    expect(clientMock.createDocument.mock.calls[0][1].kind).toBeUndefined();
-    expect(clientMock.createDocument.mock.calls[0][2]).toHaveProperty(
-      "partitionKey",
-      aFiscalCode
+    expect(containerMock.items.create).toHaveBeenCalledTimes(1);
+    expect(containerMock.items.create).toHaveBeenCalledWith(
+      expect.objectContaining({
+        kind: "INewProfile",
+        fiscalCode: aProfileWithDifferentEmail.fiscalCode,
+        email: aProfileWithDifferentEmail.email
+      }),
```

Comment:
We need to check that all the relevant properties are used to create a new version for the profile.
Most relevant are: `id` and `version`

Reply:
you are right, thanks. done in https://github.com/pagopa/io-functions-commons/pull/295/commits/7d00e44bc545338b56405a860cef3bb0c12f8659